smartsim/entity/model.py

```diff
@@ -357,21 +368,40 @@ def _set_colocated_db_settings(
             )
 
         # TODO list which db settings can be extras
-        common_options["custom_pinning"] = self._create_pinning_string(
-            common_options["custom_pinning"], common_options["cpus"]
+        x = t.cast(
+            t.Optional[t.Iterable[t.Union[int, t.Iterable[int]]]],
+            common_options.get("custom_pinning"),
         )
-
-        colo_db_config = {}
+        e = t.cast(int, common_options.get("cpus"))
+        common_options["custom_pinning"] = self._create_pinning_string(x, e)
```

Comment:
Can we change the names of these of these intermediary variables to something a bit more descriptive (maybe `custom_pinning_` and `cpus_`)?

Or maybe just remove them entirely?
```py
common_options["custom_pinning"] = self._create_pinning_string(
    t.cast(
        t.Optional[t.Iterable[t.Union[int, t.Iterable[int]]]], 
        common_options["custom_pinning"], common_options["cpus"]
    ),
    t.cast(int, common_options.get("cpus"))
)
```

Reply:
I would almost suggest keeping the vars for readability, but Im not as experienced so let me know!

Reply:
Totally willing to keep the intermediate variables! I think I agree that they improve readability!